src/main/java/org/opensearch/jobscheduler/model/ExtensionJobParameter.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.jobscheduler.model;
+
+import java.io.IOException;
+import java.time.Instant;
+
+import org.opensearch.common.Nullable;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.spi.schedule.CronSchedule;
+import org.opensearch.jobscheduler.spi.schedule.IntervalSchedule;
+import org.opensearch.jobscheduler.spi.schedule.Schedule;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+
+/**
+ * A {@link Writeable} ScheduledJobParameter used to transport job parameters between OpenSearch and Extensions
+ *
+ */
+public class ExtensionJobParameter implements ScheduledJobParameter, Writeable {
+
+    enum ScheduleType {
+        CRON,
```

Comment:
(Docs) Assuming this is API and not package private, would be helpful if these enum values had a javadoc on them describing their use, e.g., CRON is at a scheduled time of day, INTERVAL is repeating every N TimeUnits.

I assume this enum is copied from an existing implementation and these are the only two options?